### ISSUE TITLE
CSL styles are not applicable to `bibliography`

### ISSUE THREAD

**Issue by daskol** [typst/typst#2528]
### Description

I was experimenting with the latest version of `typst` and set `style` argument for `bibliography` function like below.

```typst
#bibliography(bibliography-file, style: "alphanumeric")
```

Compiler produced "CSL style is not suitable for bibliographies" while [documentation][1] says that it is possible use the same set of styles in `bibliography` function.

[1]: https://typst.app/docs/reference/meta/bibliography/

### Typst version

- [X] I am using the latest version of Typst

**Comment by laurmaedje** [comment]
Yes, cite and bibliography both use the same list although a few select styles are not applicable to bibliographies.

**Comment by laurmaedje** [comment]
Yes. But only very few ones aren't suitable for bibliographies.

**Comment by a-stevan** [comment]
i'm wondering _why_ the alphanumeric style which, i believe as per [_Bibtex bibliography styles_](https://www.overleaf.com/learn/latex/Bibtex_bibliography_styles), will look like the attached image, is considered _not suitable_ for bibliographies by Typst 🤔 

<img width="637" height="257" alt="Image" src="https://github.com/user-attachments/assets/f7801505-f644-42ee-9175-e29b21e721dd" />

**Comment by cdesaintguilhem** [comment]
**TL;DR:** using a `set` rule on `cite`, or explicitly citing alphanumerically for the _first_ citation changes the numbering in the bibliography.

I just did a little test in Typst 0.15.1 with the following code.

```typst
#set page(paper: "a5", height: auto)

#set cite(style: "alphanumeric")

We're exploring bibliographies.
Here I cite the famous RSA paper with an \@ reference~@ACM:RivShaAdl78.

#bibliography("refs.bib")
```

This produced the following result.

<img width="1748" height="815" alt="Image" src="https://github.com/user-attachments/assets/6a376ff9-d4f2-49ce-bdf5-764fa4ac7d9d" />

Without the set rule, this behaviour where the reference in the printed bibliography changes to alphanumeric also seems to happen if `style: "alphanumeric` is passed to _the first_ individual call to `cite`. If the `@` syntax is used first, the bibliography will default to "[n]", if the `@` syntax is used _after_ a parametrised call to `cite()` then the `@` citation will display as "[1]" in the text, but the bibliography will remain alphanumeric.

This was again discussed in #2707, and the behaviour I describe is a consequence of [this commit](https://github.com/typst/hayagriva/commit/22433f0cdbb3a26ddf94817c770b7b2469c08911). I'm posting this here if anyone like me lands here with the same issue in the future.

It would be nice if writing `#bibliography(sources.bib, style: "alphanumeric")` was synonymous for the above behaviour.
